timemachine/fe/single_topology.py

```diff
@@ -972,12 +1027,15 @@ def __init__(self, mol_a, mol_b, core: NDArray):
         self.c_to_a = {int(v): k for k, v in enumerate(self.a_to_c)}
         self.c_to_b = {int(v): k for k, v in enumerate(self.b_to_c)}
 
+    @cache
```

Comment:
nit: it might be worth making `AtomMapMixin` immutable (e.g. dataclass with `frozen=True`); otherwise, the cached values could become inconsistent with the instance attributes.

Reply:
added a note in docstring of `SingleTopology` for now:

"""
        Note: this class is assumed to be frozen and immutable, and changing attributes like the core
        and/or the forcefield will lead to undefined behavior.
"""